File: 2021/07/part1.pl
#!/usr/bin/perl -w

#use Data::Dumper;
#use List::Util qw/min max/;

use strict;

sub median {
  my @vals = sort {$a <=> $b} @_;
  my $len = @vals;
  if($len%2) {
    return $vals[int($len/2)];
  } else {
    return ($vals[int($len/2)-1] + $vals[int($len/2)])/2;
  }
}

sub sum_distance {
  my ($point, $arr) = @_;

  my $distance = 0;
  for my $i (@$arr) {
    $distance += abs($i - $point);
  }
  return $distance;
}

############ MAIN ##################

my $file = "input.txt";
if (scalar(@ARGV) > 0 and -e $ARGV[0]) {
  $file = $ARGV[0];
}
open(FILE, $file) or die("Cannot open file: $file\n");

my $line = <FILE>;
chomp($line);

close(FILE);

my @crabs = split(',', $line);

my $median = median(@crabs);
my $distance = sum_distance($median, \@crabs);

printf("\nResult: %d\n", $distance);

File: 2023/01/part1.py
#!/usr/bin/env python3

import re

result = 0
with open('input.txt') as f:
    for line in f:
        if not line:
            break
        #print(line.strip())
        results = re.findall("[0-9]", line)
        #print(results)
        coordonates = results[0] + "" + results[-1]
        result += int(coordonates)
        print("Numbers: %s - Coordonates: %s - Total: %s" % (results, coordonates, result))

print("===========")
print("Result: %s" % (result))

        
        
